src/tasks/types.ts

```diff
@@ -0,0 +1,58 @@
+import type { ValidateFunction } from 'ajv';
+import type { AnyObject } from 'pinejs-client-core';
+import type * as Db from '../database-layer/db';
+import type { PinejsClient } from '../sbvr-api/sbvr-utils';
+
+export const taskStatuses = [
+	'queued',
+	'cancelled',
+	'succeeded',
+	'failed',
+] as const;
+export type TaskStatus = (typeof taskStatuses)[number];
+export interface Task {
```

Comment:
Just a small reminder most of this can go aways/be auto generated once we merge: https://github.com/balena-io/pinejs/pull/763